packages/datatrak-web/src/api/mutations/useSubmitSurvey.ts

```diff
@@ -55,35 +82,54 @@ export const processSurveyResponse = ({
       continue;
     }
 
+    // base answer object to be added to the answers array
+    const answerObject = {
+      id,
+      question_id: questionId,
+      type: questionType,
+      body: answer,
+    };
+
     // Handle special question types
     // TODO: add in photo and file upload handling, as well as adding new entities when these question types are implemented
     switch (questionType) {
       // format dates to be ISO strings
       case 'SubmissionDate':
       case 'DateOfData':
-        surveyResponseData.data_time = moment(answer).toISOString();
+        surveyResponse.data_time = moment(answer as string).toISOString();
         break;
       // add the entity id to the response if the question is a primary entity question
       case 'PrimaryEntity': {
-        surveyResponseData.entity_id = answer;
+        surveyResponse.entity_id = answer as string;
         break;
       }
-      default:
+      case 'Autocomplete': {
+        // if the answer is a new option, add it to the options_created array to be added to the DB
+        const { isNew, value, label, optionSetId } = answer as AutocompleteAnswer;
+        if (isNew) {
+          surveyResponse.options_created.push({
+            option_set_id: optionSetId,
```

Comment:
I noticed that on Meditrak the optionsCreated have an id and a sortOrder, does it need them here too?

Reply:
I don't think so - it automatically adds the sort order as the the next number (to make the option the last one in the list) and a unique id gets applied when it's created